policy-controller/k8s/status/src/index.rs

```diff
@@ -74,14 +111,41 @@ impl Index {
         updates: UnboundedSender<Update>,
     ) -> SharedIndex {
         Arc::new(RwLock::new(Self {
-            _name: name.to_string(),
-            _claims: claims,
+            name: name.to_string(),
+            claims,
             updates,
             http_routes: HashMap::new(),
             servers: HashSet::new(),
         }))
     }
 
+    pub async fn run(index: Arc<RwLock<Self>>) {
+        // Clone the claims watch out of the index. This will immediately
+        // drop the read lock on the index so that it is not held for the
+        // lifetime of this function.
+        let mut claims = index.read().claims.clone();
+
+        loop {
+            tokio::select! {
+                _ = claims.changed() => {
+                    tracing::debug!("Lease holder has changed")
+                }
+                _ = tokio::time::sleep(std::time::Duration::from_secs(10)) => {}
+            }
+
+            // The claimant has changed or we should attempt to reconcile all
+            // HTTPRoutes to account for any errors. In either case, we should
+            // only proceed if we are the current leader.
+            let claims = claims.borrow_and_update();
+            let index = index.write();
+            if !claims.is_current_for(&index.name) {
+                continue;
+            }
+            tracing::debug!(%index.name, "Lease holder reconciling cluster");
+            Index::reconcile(&*index);
```

Comment:
nit, not important: i think we ought to be able to call this as just
```suggestion
            index.reconcile();
```
since rust should automatically dereference the receiver...does that not work?